ios/TransactionListView.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  TransactionListView.swift
+//  Rainbow
+//
+//  Created by Alexey Kureev on 28/12/2019.
+//
+
+import Foundation
+
+fileprivate struct TransactionSection {
+  var header: Date
+  var transactions: [Transaction]
+}
+
+class TransactionListView: UIView {
+  @objc lazy var onItemPress: RCTBubblingEventBlock = { _ in }
+  @objc lazy var onReceivePress: RCTBubblingEventBlock = { _ in }
+  @objc lazy var onCopyAddressPress: RCTBubblingEventBlock = { _ in }
+  @objc var duration: TimeInterval = 0.15
+  @objc var scaleTo: CGFloat = 0.97
+  @objc var enableHapticFeedback: Bool = true
+  @objc var hapticType: String = "selection"
+  @objc var accountAddress: String? = nil {
+    didSet {
+      header.accountAddress.text = accountAddress
+    }
+  }
+  @objc var transactions: [Transaction] = [] {
+    /// Every time we receive a new set of transactions, regroup by mind_at in the format "MMMM yyyy"
+    /// Then, re-render tableView with the new data
+    didSet {
+      var groups: [Date: [Transaction]] = [:]
+      let calendar = Calendar.current
+      
+      for transaction in transactions {
+        var date = groupByDate(transaction.minedAt)
+        
+        if calendar.isDateInToday(date) || calendar.isDateInYesterday(date) {
+          if groups[date] == nil {
+            groups[date] = []
+          }
+          groups[date]!.append(transaction)
+        } else {
+          let dateComponents = calendar.dateComponents([.year, .month], from: date)
+          date = calendar.date(from: dateComponents)!
+          
+          if groups[date] == nil {
+            groups[date] = []
+          }
+          
+          groups[date]!.append(transaction)
+        }
+      }
+      
+      sections = groups.map(TransactionSection.init(header:transactions:))
+      sections.sort { (lhs, rhs) in lhs.header > rhs.header }
+      tableView.reloadData()
+    }
+  }
+  @objc func onReceivePressed(_ sender: UIButton) {
+    self.onReceivePress([:])
+  }
+  @objc func onCopyAddressPressed(_ sender: UIButton) {
+    let rect = sender.convert(sender.frame, to: self)
+    self.onCopyAddressPress([
+      "x": rect.origin.x,
+      "y": rect.origin.y,
+      "width": sender.frame.width,
+      "height": sender.frame.height
+    ])
+  }
+  
+  fileprivate var sections = [TransactionSection]()
+  
+  let tableView = UITableView()
+  let header: TransactionListViewHeader = TransactionListViewHeader.fromNib()
+  let headerSeparator = UIView()
+  
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    
+    tableView.dataSource = self
+    tableView.delegate = self
+    tableView.rowHeight = 60
+    tableView.delaysContentTouches = false
+    tableView.separatorStyle = UITableViewCell.SeparatorStyle.none
+    tableView.register(UINib(nibName: "TransactionListViewCell", bundle: nil), forCellReuseIdentifier: "TransactionListViewCell")
```

Comment:
it's better to move `"TransactionListViewCell"` to a constant since you're using it for cell as well.

Reply:
You mean the string, right?